src/test/java/com/google/sps/servlets/AddSchoolDataTest.java

```diff
@@ -0,0 +1,449 @@
+            .get(0)
+            .getKey();
+    List<Entity> children = findChildren(db, "Course", school);
+
+    assertEquals(2, children.size());
+  }
+
+  @Test
+  public void AddingNewTerms_SameCourseAncestor() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+    when(request.getParameter("school-name")).thenReturn("MIT");
+    when(request.getParameter("course-name")).thenReturn("6.006");
+    when(request.getParameter("term")).thenReturn("Spring 2020");
+    when(request.getParameter("units")).thenReturn("12");
+    when(request.getParameter("professor-name")).thenReturn("Jason Ku");
+    newSchoolObject.addSchoolData(db, request);
+
+    requestB = Mockito.mock(HttpServletRequest.class);
+    when(requestB.getParameter("school-name")).thenReturn("MIT");
+    when(requestB.getParameter("course-name")).thenReturn("6.006");
+    when(requestB.getParameter("term")).thenReturn("Spring 2021");
+    when(requestB.getParameter("units")).thenReturn("12");
+    when(requestB.getParameter("professor-name")).thenReturn("Srini");
+    newSchoolObject.addSchoolData(db, requestB);
+
+    Key course =
+        findQueryMatch(db, "Course", "course-name", request.getParameter("course-name"), false)
+            .get(0)
+            .getKey();
+    List<Entity> children = findChildren(db, "Term", course);
+
+    assertEquals(2, children.size());
+  }
+
+  @Test
+  public void AddingSchoolProfessors_SameSchoolAncestor() {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+    when(request.getParameter("school-name")).thenReturn("MIT");
+    when(request.getParameter("course-name")).thenReturn("6.006");
+    when(request.getParameter("term")).thenReturn("Spring 2020");
+    when(request.getParameter("units")).thenReturn("12");
+    when(request.getParameter("professor-name")).thenReturn("Jason Ku");
+    newSchoolObject.addSchoolData(db, request);
+
+    requestB = Mockito.mock(HttpServletRequest.class);
+    when(requestB.getParameter("school-name")).thenReturn("MIT");
+    when(requestB.getParameter("course-name")).thenReturn("6.008");
+    when(requestB.getParameter("term")).thenReturn("Spring 2021");
+    when(requestB.getParameter("units")).thenReturn("12");
+    when(requestB.getParameter("professor-name")).thenReturn("Srini");
+    newSchoolObject.addSchoolData(db, requestB);
+
+    Key school =
+        findQueryMatch(db, "School", "school-name", request.getParameter("school-name"), false)
+            .get(0)
+            .getKey();
+    List<Entity> children = findChildren(db, "Professor", school);
+
+    assertEquals(2, children.size());
+  }
+
+  public List<Entity> findQueryMatch(
+      DatastoreService db,
+      String entityType,
+      String entityProperty,
+      String propertyValue,
+      Boolean onlyType) {
```

Comment:
If this boolean is true, do you nee entityProperty and propertyValue? Seems like you could make those both optional parameters here, or overload your function here for the two different query patterns.